pkg/operator/bootstrap.go

```diff
@@ -52,28 +64,74 @@ func (h *Handler) Bootstrap() (*regopapi.ImageRegistry, error) {
 		return nil, fmt.Errorf("only one registry custom resource expected in %s namespace, got %d", h.params.Deployment.Namespace, len(crList.Items))
 	}
 
-	logrus.Infof("generating registry custom resource")
+	var spec imageregistryapi.ImageRegistrySpec
 
 	// TODO(legion): Add real bootstrap based on global ConfigMap or something.
-	cr := &regopapi.ImageRegistry{
+	dc := &appsapi.DeploymentConfig{
 		TypeMeta: metav1.TypeMeta{
-			APIVersion: regopapi.SchemeGroupVersion.String(),
-			Kind:       "ImageRegistry",
+			APIVersion: appsapi.SchemeGroupVersion.String(),
+			Kind:       "DeploymentConfig",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:       "image-registry",
-			Namespace:  h.params.Deployment.Namespace,
-			Finalizers: []string{parameters.ImageRegistryOperatorResourceFinalizer},
+			Name:      resourceName(h.params.Deployment.Namespace),
+			Namespace: h.params.Deployment.Namespace,
 		},
-		Spec: regopapi.ImageRegistrySpec{
+	}
+	if err := sdk.Get(dc); err != nil {
```

Comment:
this should distinguish between a not found error and other types of errors so we don't create a new registry in cases where we had an api failure checking for an existing one.